File: 프로그래머스/lv1/12948. 핸드폰 번호 가리기/핸드폰 번호 가리기.js
function solution(phone_number) {
    // 전방 탐색 기호는 ?= 이며, = 다음에 오는 문자가 일치하는 영역에서 제외
    // 또한, 전방 탐색은 하위 표현식과 같이 소괄호로 감싸주어야만 합니다
    // return phone_number.replace(/\d(?=\d{4})/g, "*")
    // return "*".repeat(phone_number.length-4) + phone_number.slice(-4)
    
    return phone_number.slice(-4).padStart(phone_number.length, "*")
}

File: 프로그래머스/lv1/12948. 핸드폰 번호 가리기/핸드폰 번호 가리기.java
class Solution {
    public String solution(String phone_number) {
        // String.toCharArray()는 문자열을 한 글자씩 쪼개서
        // 이를 char타입의 배열에 집어넣어주는 친절한 메소드
        char[] arr = phone_number.toCharArray();
        
        for(int i = 0; i < arr.length - 4; i++) {
            arr[i] = '*';
        }
        // valueOf() 함수는 입력받은 인자 값을 지정된
        // Number 객체 형으로 변환하여 반환
        return String.valueOf(arr);
    }
}